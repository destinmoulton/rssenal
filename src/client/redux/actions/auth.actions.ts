import { API_AUTH } from "../apiendpoints";
import { JSON_HEADERS } from "../../lib/headers";

import { IDispatch } from "../../interfaces";

export function authenticateUser(username: string, password: string) {
    return (dispatch: IDispatch) => {
        const url = API_AUTH;
        const init = {
            method: "POST",
            body: JSON.stringify({ username, password }),
            headers: JSON_HEADERS
        };

        fetch(url, init)
            .then(res => {
                return res.json();
            })
            .then(auth => {
                if (auth.isAuthorized) {
                }
            })
            .catch(err => {});
    };
}
