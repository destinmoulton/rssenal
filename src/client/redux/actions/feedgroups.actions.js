import {
    FEEDGROUPS_FETCHING,
    FEEDGROUPS_RECEIVED,
    FEEDGROUPS_UPDATE_BEGIN,
    FEEDGROUPS_UPDATE_COMPLETE
} from "../actiontypes";

import {
    API_FEEDGROUPS_BASE,
    API_FEEDGROUPS_GET_ALL
} from "../apiendpoints";

export function getAllFeedGroups(){
    return (dispatch)=>{
        dispatch(fetchingInProgress());
        dispatch(fetchFeedGroups());
    }
}

function fetchingInProgress(){
    return {
        type: FEEDGROUPS_FETCHING
    }
}

function fetchFeedGroups(){
    return (dispatch)=>{
        const url = API_FEEDGROUPS_GET_ALL;
        const init = {
            method: "GET"
        };
        fetch(url, init)
            .then((res)=>{
                return res.json();
            })
            .then((groups)=>{
                dispatch(fetchingComplete(groups));
            })
    }
}

function fetchingComplete(data){
    return {
        type: FEEDGROUPS_RECEIVED,
        groups: data
    }
}

export function beginSaveFeedGroup(groupId, newGroupName){
    return (dispatch)=>{
        dispatch(updatingInProgress(groupId));
        dispatch(updateFeedGroup(groupId, newGroupName));
    };
}

function updatingInProgress(groupId){
    return {
        type: FEEDGROUPS_UPDATE_BEGIN,
        groupId
    };
}

function updateFeedGroup(groupId, newGroupName){
    return (dispatch)=>{
        const url = API_FEEDGROUPS_BASE + groupId;
        const init = {
            method: "PUT",
            body: JSON.stringify({name: newGroupName}),
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        };

        fetch(url, init)
            .then((res)=>{
                return res.json();
            })
            .then((newGroup)=>{
                dispatch(updatingComplete(newGroup))
            })
    }
}

function updatingComplete(newGroup){
    return {
        type: FEEDGROUPS_UPDATE_COMPLETE,
        group: newGroup
    }
}