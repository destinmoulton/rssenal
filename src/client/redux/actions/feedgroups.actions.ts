import {
    FEEDGROUPS_FETCHING,
    FEEDGROUPS_RECEIVED,
    FEEDGROUPS_ADD_BEGIN,
    FEEDGROUPS_ADD_COMPLETE,
    FEEDGROUPS_DELETE_BEGIN,
    FEEDGROUPS_DELETE_COMPLETE,
    FEEDGROUPS_UPDATE_BEGIN,
    FEEDGROUPS_UPDATE_COMPLETE
} from "../actiontypes";

import {
    API_FEEDGROUPS_BASE,
    API_FEEDGROUPS_GET_ALL
} from "../apiendpoints";

import { IDispatch, IFeedgroup, TFeedgroupID } from "../../interfaces";

export function getAllFeedGroups(){
    return (dispatch: IDispatch)=>{
        dispatch(fetchingInProgress());
        dispatch(fetchFeedGroups());
    }
}

function fetchingInProgress(){
    return {
        type: FEEDGROUPS_FETCHING
    }
}

function fetchFeedGroups(){
    return (dispatch: IDispatch)=>{
        const url = API_FEEDGROUPS_GET_ALL;
        const init = {
            method: "GET"
        };
        fetch(url, init)
            .then((res)=>{
                return res.json();
            })
            .then((groups)=>{
                dispatch(fetchingComplete(groups));
            })
    }
}

function fetchingComplete(data: IFeedgroup[]){
    return {
        type: FEEDGROUPS_RECEIVED,
        groups: data
    }
}

export function beginSaveFeedGroup(groupInfo: IFeedgroup){
    return (dispatch: IDispatch)=>{
        if(groupInfo._id !== ""){
            dispatch(updatingInProgress());
            dispatch(updateFeedGroup(groupInfo._id, groupInfo.name));
        } else {
            dispatch(addingInProgress());
            dispatch(addFeedGroup(groupInfo.name));
        }
    }
}

function addingInProgress(){
    return {
        type: FEEDGROUPS_ADD_BEGIN
    }
}

function addFeedGroup(newGroupName: string){
    return (dispatch: IDispatch)=>{
        const url = API_FEEDGROUPS_BASE;
        const init = {
            method: "POST",
            body: JSON.stringify({name: newGroupName}),
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        }
        fetch(url, init)
            .then((res)=>{
                return res.json();
            })
            .then((newGroup)=>{
                dispatch(addingFeedGroupComplete(newGroup));
            })
    }
}

function addingFeedGroupComplete(group: IFeedgroup){
    return {
        type: FEEDGROUPS_ADD_COMPLETE,
        group
    };
}

function updatingInProgress(){
    return {
        type: FEEDGROUPS_UPDATE_BEGIN
    };
}

function updateFeedGroup(groupId: TFeedgroupID, newGroupName: string){
    return (dispatch: IDispatch)=>{
        const url = API_FEEDGROUPS_BASE + groupId;
        const init = {
            method: "PUT",
            body: JSON.stringify({name: newGroupName}),
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        };

        fetch(url, init)
            .then((res)=>{
                return res.json();
            })
            .then((newGroup)=>{
                dispatch(updatingComplete(newGroup))
            })
    }
}

function updatingComplete(newGroup: IFeedgroup){
    return {
        type: FEEDGROUPS_UPDATE_COMPLETE,
        group: newGroup
    }
}

export function beginDeleteFeedGroup(groupId: TFeedgroupID){
    return (dispatch: IDispatch)=>{
        dispatch(deleteInProgress());
        dispatch(deleteFeedGroup(groupId));
    }
}

function deleteInProgress(){
    return {
        type: FEEDGROUPS_DELETE_BEGIN
    }
}

function deleteFeedGroup(groupId: TFeedgroupID){
    return (dispatch: IDispatch)=>{
        const url = API_FEEDGROUPS_BASE + groupId;
        const init = {
            method: "DELETE"
        }
        fetch(url, init)
            .then((res)=>{
                return res.json();
            })
            .then((resObj)=>{
                if(resObj.status==="success"){
                    dispatch(deleteComplete(groupId));
                }
            })
    }
}

function deleteComplete(groupId: TFeedgroupID){
    return {
        type: FEEDGROUPS_DELETE_COMPLETE,
        groupId
    }
}