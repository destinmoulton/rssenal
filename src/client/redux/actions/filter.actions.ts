import { OrderedMap } from "immutable";

import { FILTER_CHANGE, FILTER_RESET } from "../actiontypes";
import * as Types from "../../types";

export function filterChangeActive(filter: Types.IFilter) {
    return (dispatch: Types.IDispatch, getState: Types.IGetState) => {
        const { entriesStore } = getState();
        dispatch(filterVisibleEntries(filter, entriesStore.entries));
    };
}

export function filterVisibleEntries(
    filter: Types.IFilter,
    currentEntries: Types.TEntries
) {
    return (dispatch: Types.IDispatch, getState: Types.IGetState) => {
        const { feedsStore, foldersStore } = getState();
        const allFeeds = feedsStore.feeds;

        let title = "All";

        let filteredEntries = currentEntries.toOrderedMap();
        switch (filter.limit) {
            case "feed":
                const activeFeed = feedsStore.feeds.get(filter.id);
                title = activeFeed.title;

                filteredEntries = currentEntries
                    .filter(entry => {
                        return entry.feed_id === filter.id;
                    })
                    .toOrderedMap();
                break;
            case "folder":
                if (filter.id !== "all") {
                    const activeFolder = foldersStore.folders.get(filter.id);

                    title = activeFolder.name;

                    const feedIds = allFeeds
                        .filter(feed => {
                            return feed.folder_id === filter.id;
                        })
                        .map(feed => {
                            return feed._id;
                        });

                    filteredEntries = currentEntries
                        .filter(entry => {
                            return feedIds.includes(entry.feed_id);
                        })
                        .toOrderedMap();
                }
                break;
        }
        dispatch(setNewFilter(filter, title, filteredEntries));
    };
}

function setNewFilter(
    newFilter: Types.IFilter,
    filterTitle: string,
    filteredEntries: OrderedMap<Types.TEntryID, Types.IEntry>
) {
    return {
        type: FILTER_CHANGE,
        newFilter,
        filterTitle,
        filteredEntries
    };
}

export function filterReset() {
    return {
        type: FILTER_RESET
    };
}
