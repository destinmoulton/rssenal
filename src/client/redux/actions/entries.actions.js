
import {
    ENTRIES_GET_BEGIN,
    ENTRIES_GET_COMPLETE,
    ENTRIES_UPDATE_BEGIN,
    ENTRIES_UPDATE_COMPLETE
} from "../actiontypes";

import {
    API_ENTRIES_BASE
} from "../apiendpoints";

import { feedsDecrementUnread, feedsSetAllUnreadCount } from "./feeds.actions";

export function beginGetEntries(){
    return (dispatch)=>{
        dispatch(beginGetProcess());
        dispatch(getEntries());
    }
}

function beginGetProcess(){
    return {
        type: ENTRIES_GET_BEGIN
    }
}

function getEntries(){
    return (dispatch, getState)=>{
        const url = API_ENTRIES_BASE;
        const init = {
            method: "GET"
        };

        fetch(url, init)
            .then((res)=>{
                return res.json();
            })
            .then((resObj)=>{
                if(resObj.status === "error"){
                    console.error(resObj.error);
                } else {
                    let entries = resObj.entries;
                    dispatch(getEntriesComplete(entries));
                    dispatch(entriesUpdateUnreadCount(entries));
                }
            })
    }
}

function getEntriesComplete(entries){
    return {
        type: ENTRIES_GET_COMPLETE,
        entries
    }
}

function entriesUpdateUnreadCount(entries){
    return (dispatch)=>{
        dispatch(feedsSetAllUnreadCount(entries));
    }
}

export function updateReadState(entry, hasRead){
    return (dispatch)=>{
        const url = API_ENTRIES_BASE + entry._id;
        const init = {
            method: "PUT",
            body: JSON.stringify({has_read: hasRead}),
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        };

        fetch(url, init)
            .then((res)=>{
                return res.json();
            })
            .then((resObj)=>{
                if(resObj.status === "error"){
                    console.error(resObj.error);
                } else {
                    dispatch(feedsDecrementUnread(entry.feed_id));
                }
            });
    }
}