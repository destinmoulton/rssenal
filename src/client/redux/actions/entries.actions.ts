import { OrderedMap } from "immutable";
import * as moment from "moment";

import { ENTRIES_CLEAR_ALL, ENTRIES_SET_ALL } from "../actiontypes";

import { API_ENTRIES_BASE } from "../apiendpoints";

import { generateJWTJSONHeaders, generateJWTHeaders } from "../../lib/headers";
import { feedsDecrementUnread, feedsSetAllUnreadCount } from "./feeds.actions";
import { filterVisibleEntries } from "./filter.actions";
import { message } from "./messages.actions";

import {
    IDispatch,
    IEntry,
    IGetState,
    TFeedID,
    TEntryID,
    TEntries
} from "../../interfaces";

export function getEntriesForFeed(feedId: TFeedID) {
    return (dispatch: IDispatch, getState: IGetState) => {
        const { settingsStore } = getState();
        let showRead = false;
        settingsStore.settings.forEach(setting => {
            if (setting.key === "show_entries_has_read") {
                showRead = setting.value;
            }
        });

        const queryString =
            "?showEntriesHasRead=" + showRead + "&feedId=" + feedId;
        const url = API_ENTRIES_BASE + queryString;
        const init = {
            method: "GET",
            headers: generateJWTHeaders()
        };

        fetch(url, init)
            .then(res => {
                return res.json();
            })
            .then(resObj => {
                if (resObj.status === "error") {
                    dispatch(message(resObj.error, "error"));
                    console.error(resObj.error);
                } else {
                    let entries = resObj.entries;
                    dispatch(ammendEntries(entries));
                }
            })
            .catch(err => {
                console.error(err);
            });
    };
}

function ammendEntries(entries: IEntry[]) {
    return (dispatch: IDispatch, getState: IGetState) => {
        const { feedsStore, filterStore } = getState();

        const ammendedEntries = entries.map((entry: IEntry) => {
            const feedTitle = feedsStore.feeds.get(entry.feed_id).title;
            const timeAgo = moment(entry.publish_date).fromNow();
            return {
                ...entry,
                feedTitle,
                timeAgo
            };
        });
        dispatch(getEntriesComplete(ammendedEntries));

        dispatch(feedsSetAllUnreadCount(ammendedEntries));
    };
}

function getEntriesComplete(entries: IEntry[]) {
    return (dispatch: IDispatch, getState: IGetState) => {
        const { entriesStore, filterStore } = getState();
        const currentEntries = entriesStore.entries;

        let newEntries = currentEntries.toOrderedMap();
        entries.forEach(entry => {
            newEntries = newEntries.set(entry._id, entry);
        });

        dispatch(entriesSetAndFilter(newEntries));
    };
}

export function updateReadState(entry: IEntry, hasRead: boolean) {
    return (dispatch: IDispatch) => {
        const url = API_ENTRIES_BASE + entry._id;
        const init = {
            method: "PUT",
            body: JSON.stringify({ has_read: hasRead }),
            headers: generateJWTJSONHeaders()
        };

        fetch(url, init)
            .then(res => {
                return res.json();
            })
            .then(resObj => {
                if (resObj.status === "error") {
                    console.error(resObj.error);
                } else {
                    dispatch(entryAmmendMarkRead(entry._id));
                }
            })
            .catch(err => {
                console.error(err);
            });
    };
}

function entryAmmendMarkRead(entryId: TEntryID) {
    return (dispatch: IDispatch, getState: IGetState) => {
        const { entriesStore, filterStore } = getState();
        const allEntries = entriesStore.entries;
        const entry = allEntries.get(entryId);

        const newEntry = { ...entry, has_read: true };
        const newEntries = allEntries.set(entryId, newEntry);

        dispatch(feedsDecrementUnread(entry.feed_id));
        dispatch(entriesSetAndFilter(newEntries));
    };
}

export function entriesRemoveFeed(feedId: TFeedID) {
    return (dispatch: IDispatch, getState: IGetState) => {
        const { entriesStore } = getState();
        const { entries } = entriesStore;

        const newEntries = entries
            .filter((entry: IEntry) => {
                return entry.feed_id !== feedId;
            })
            .toOrderedMap();
        dispatch(entriesSetAndFilter(newEntries));
    };
}

function entriesSetAndFilter(entries: TEntries) {
    return (dispatch: IDispatch, getState: IGetState) => {
        const { filterStore } = getState();
        dispatch(entriesSetAll(entries));
        dispatch(filterVisibleEntries(filterStore.filter, entries));
    };
}

function entriesSetAll(entries: TEntries) {
    return {
        type: ENTRIES_SET_ALL,
        entries
    };
}

export function entriesClearAll() {
    return {
        type: ENTRIES_CLEAR_ALL
    };
}
