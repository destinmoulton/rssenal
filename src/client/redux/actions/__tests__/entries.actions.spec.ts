import * as fetchMock from "fetch-mock";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

import * as EntriesActions from "../entries.actions";
import * as ACT_TYPES from "../../actiontypes";
import * as INIT_STATE from "../../initialstate";

import API_ENTRIES_STRING from "../../../../../test/data/api/entries.unread";
import AMMENDED_ENTRIES from "../../../../../test/data/immutable/ammendedEntries";
import IMM_FEEDS from "../../../../../test/data/immutable/feeds";
import IMM_UNREAD_MAP from "../../../../../test/data/immutable/unreadMap";

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

describe("entries.actions", () => {
    afterEach(() => {
        fetchMock.reset();
        fetchMock.restore();
        localStorage.clear();
    });

    it("entriesGetAllForFeed() fetches and ammends entries", () => {
        const feedID = "5b33c76cb2438d5708dc197e";
        const url = "/api/entries/?showEntriesHasRead=false&feedId=" + feedID;

        fetchMock.getOnce(url, JSON.parse(API_ENTRIES_STRING));

        const expectedActions = [
            { type: ACT_TYPES.ENTRIES_SET_ALL, entries: AMMENDED_ENTRIES },
            {
                type: ACT_TYPES.FILTER_CHANGE,
                filterTitle: "All",
                newFilter: { id: "all", limit: "folder" },
                filteredEntries: AMMENDED_ENTRIES
            },
            { type: ACT_TYPES.FEEDS_SET_UNREAD, unreadMap: IMM_UNREAD_MAP }
        ];

        const store = mockStore({
            entriesStore: INIT_STATE.ENTRIES_INITIAL_STATE,
            feedsStore: {
                ...INIT_STATE.FEEDS_INITIAL_STATE,
                feeds: IMM_FEEDS
            },
            filterStore: INIT_STATE.FILTER_INITIAL_STATE,
            settingsStore: INIT_STATE.SETTINGS_INITIAL_STATE
        });

        return store
            .dispatch(EntriesActions.entriesGetAllForFeed(feedID))
            .then(() => {
                expect(store.getActions()).toEqual(expectedActions);
            });
    });
});
