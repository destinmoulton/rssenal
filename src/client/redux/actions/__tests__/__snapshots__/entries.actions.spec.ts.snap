// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entries.actions entriesClearAll() clears the current entries 1`] = `
Array [
  Object {
    "entries": Immutable.OrderedMap {},
    "type": "ENTRIES_SET_ALL",
  },
  Object {
    "filterTitle": "All",
    "filteredEntries": Immutable.OrderedMap {},
    "type": "FILTER_SET_DATA",
  },
]
`;

exports[`entries.actions entriesGetForFeed() fetches and ammends entries 1`] = `
Array [
  Object {
    "entries": Immutable.OrderedMap {
      "5b33c76cb2438d5708dc197f": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc197f",
        "content": "<p>Content for Stuff The Internet Says On Scalability For June 22nd, 2018</p>",
        "content_snippet": "<p>Content for Stuff The Internet Says On Scalability For June 22nd, 2018</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36082622",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/uqbrISxcA-E/stuff-the-internet-says-on-scalability-for-june-22nd-2018.html",
        "publish_date": "2018-06-22T15:56:46.000Z",
        "timeAgo": "21 days ago",
        "title": "Stuff The Internet Says On Scalability For June 22nd, 2018",
      },
      "5b33c76cb2438d5708dc1981": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1981",
        "content": "<p>Content for Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL</p>",
        "content_snippet": "<p>Content for Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36081782",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/gKe2fkOoT0I/sponsored-post-datadog-inmemorynet-triplebyte-etleap-scalyr.html",
        "publish_date": "2018-06-19T16:10:13.000Z",
        "timeAgo": "24 days ago",
        "title": "Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL",
      },
      "5b33c76cb2438d5708dc1983": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1983",
        "content": "<p>Content for How Ably Efficiently Implemented Consistent Hashing</p>",
        "content_snippet": "<p>Content for How Ably Efficiently Implemented Consistent Hashing</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36081325",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/EU6Dy48k59o/how-ably-efficiently-implemented-consistent-hashing.html",
        "publish_date": "2018-06-18T16:06:17.000Z",
        "timeAgo": "25 days ago",
        "title": "How Ably Efficiently Implemented Consistent Hashing",
      },
      "5b33c76cb2438d5708dc1985": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1985",
        "content": "<p>Content for June 15th, 2018</p>",
        "content_snippet": "<p>Content for June 15th, 2018</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36079455",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/sacuGsRa2G0/stuff-the-internet-says-on-scalability-for-june-15th-2018.html",
        "publish_date": "2018-06-15T15:57:00.000Z",
        "timeAgo": "a month ago",
        "title": "Stuff The Internet Says On Scalability For June 15th, 2018",
      },
      "5b33c76cb2438d5708dc1987": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1987",
        "content": "<p>Content for Open Source Database HA Resources from Severalnines</p>",
        "content_snippet": "<p>Content for Open Source Database HA Resources from Severalnines</p>",
        "creator": "Forrest Lymburner",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36078004",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/FFJQudxO2bU/open-source-database-ha-resources-from-severalnines.html",
        "publish_date": "2018-06-08T19:46:22.000Z",
        "timeAgo": "a month ago",
        "title": "Open Source Database HA Resources from Severalnines",
      },
    },
    "type": "ENTRIES_SET_ALL",
  },
  Object {
    "filterTitle": "All",
    "filteredEntries": Immutable.OrderedMap {},
    "type": "FILTER_SET_DATA",
  },
  Object {
    "type": "FEEDS_SET_UNREAD",
    "unreadMap": Object {
      "entriesCounted": Immutable.Set [
        "5b33c76cb2438d5708dc197f",
        "5b33c76cb2438d5708dc1981",
        "5b33c76cb2438d5708dc1983",
        "5b33c76cb2438d5708dc1985",
        "5b33c76cb2438d5708dc1987",
      ],
      "feeds": Immutable.Map {
        "5b33c76cb2438d5708dc197e": 5,
      },
      "folders": Immutable.Map {
        "5b33c57eb2438d5708dc1968": 5,
      },
    },
  },
]
`;

exports[`entries.actions entryUpdateHasRead() marks an entry read 1`] = `
Array [
  Object {
    "type": "FEEDS_SET_UNREAD",
    "unreadMap": Object {
      "entriesCounted": Immutable.Set [
        "5b33c76cb2438d5708dc197f",
        "5b33c76cb2438d5708dc1981",
        "5b33c76cb2438d5708dc1983",
        "5b33c76cb2438d5708dc1985",
        "5b33c76cb2438d5708dc1987",
      ],
      "feeds": Immutable.Map {
        "5b33c76cb2438d5708dc197e": 4,
      },
      "folders": Immutable.Map {
        "5b33c57eb2438d5708dc1968": 4,
      },
    },
  },
  Object {
    "entries": Immutable.OrderedMap {
      "5b33c76cb2438d5708dc197f": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc197f",
        "content": "<p>Content for Stuff The Internet Says On Scalability For June 22nd, 2018</p>",
        "content_snippet": "<p>Content for Stuff The Internet Says On Scalability For June 22nd, 2018</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36082622",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/uqbrISxcA-E/stuff-the-internet-says-on-scalability-for-june-22nd-2018.html",
        "publish_date": "2018-06-22T15:56:46.000Z",
        "timeAgo": "21 days ago",
        "title": "Stuff The Internet Says On Scalability For June 22nd, 2018",
      },
      "5b33c76cb2438d5708dc1981": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1981",
        "content": "<p>Content for Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL</p>",
        "content_snippet": "<p>Content for Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36081782",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/gKe2fkOoT0I/sponsored-post-datadog-inmemorynet-triplebyte-etleap-scalyr.html",
        "publish_date": "2018-06-19T16:10:13.000Z",
        "timeAgo": "24 days ago",
        "title": "Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL",
      },
      "5b33c76cb2438d5708dc1983": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1983",
        "content": "<p>Content for How Ably Efficiently Implemented Consistent Hashing</p>",
        "content_snippet": "<p>Content for How Ably Efficiently Implemented Consistent Hashing</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36081325",
        "has_read": true,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/EU6Dy48k59o/how-ably-efficiently-implemented-consistent-hashing.html",
        "publish_date": "2018-06-18T16:06:17.000Z",
        "timeAgo": "25 days ago",
        "title": "How Ably Efficiently Implemented Consistent Hashing",
      },
      "5b33c76cb2438d5708dc1985": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1985",
        "content": "<p>Content for June 15th, 2018</p>",
        "content_snippet": "<p>Content for June 15th, 2018</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36079455",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/sacuGsRa2G0/stuff-the-internet-says-on-scalability-for-june-15th-2018.html",
        "publish_date": "2018-06-15T15:57:00.000Z",
        "timeAgo": "a month ago",
        "title": "Stuff The Internet Says On Scalability For June 15th, 2018",
      },
      "5b33c76cb2438d5708dc1987": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1987",
        "content": "<p>Content for Open Source Database HA Resources from Severalnines</p>",
        "content_snippet": "<p>Content for Open Source Database HA Resources from Severalnines</p>",
        "creator": "Forrest Lymburner",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36078004",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/FFJQudxO2bU/open-source-database-ha-resources-from-severalnines.html",
        "publish_date": "2018-06-08T19:46:22.000Z",
        "timeAgo": "a month ago",
        "title": "Open Source Database HA Resources from Severalnines",
      },
    },
    "type": "ENTRIES_SET_ALL",
  },
  Object {
    "filterTitle": "All",
    "filteredEntries": Immutable.OrderedMap {
      "5b33c76cb2438d5708dc197f": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc197f",
        "content": "<p>Content for Stuff The Internet Says On Scalability For June 22nd, 2018</p>",
        "content_snippet": "<p>Content for Stuff The Internet Says On Scalability For June 22nd, 2018</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36082622",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/uqbrISxcA-E/stuff-the-internet-says-on-scalability-for-june-22nd-2018.html",
        "publish_date": "2018-06-22T15:56:46.000Z",
        "timeAgo": "21 days ago",
        "title": "Stuff The Internet Says On Scalability For June 22nd, 2018",
      },
      "5b33c76cb2438d5708dc1981": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1981",
        "content": "<p>Content for Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL</p>",
        "content_snippet": "<p>Content for Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36081782",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/gKe2fkOoT0I/sponsored-post-datadog-inmemorynet-triplebyte-etleap-scalyr.html",
        "publish_date": "2018-06-19T16:10:13.000Z",
        "timeAgo": "24 days ago",
        "title": "Sponsored Post: Datadog, InMemory.Net, Triplebyte, Etleap, Scalyr, MemSQL",
      },
      "5b33c76cb2438d5708dc1983": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1983",
        "content": "<p>Content for How Ably Efficiently Implemented Consistent Hashing</p>",
        "content_snippet": "<p>Content for How Ably Efficiently Implemented Consistent Hashing</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36081325",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/EU6Dy48k59o/how-ably-efficiently-implemented-consistent-hashing.html",
        "publish_date": "2018-06-18T16:06:17.000Z",
        "timeAgo": "25 days ago",
        "title": "How Ably Efficiently Implemented Consistent Hashing",
      },
      "5b33c76cb2438d5708dc1985": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1985",
        "content": "<p>Content for June 15th, 2018</p>",
        "content_snippet": "<p>Content for June 15th, 2018</p>",
        "creator": "Todd Hoff",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36079455",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/sacuGsRa2G0/stuff-the-internet-says-on-scalability-for-june-15th-2018.html",
        "publish_date": "2018-06-15T15:57:00.000Z",
        "timeAgo": "a month ago",
        "title": "Stuff The Internet Says On Scalability For June 15th, 2018",
      },
      "5b33c76cb2438d5708dc1987": Object {
        "__v": 0,
        "_id": "5b33c76cb2438d5708dc1987",
        "content": "<p>Content for Open Source Database HA Resources from Severalnines</p>",
        "content_snippet": "<p>Content for Open Source Database HA Resources from Severalnines</p>",
        "creator": "Forrest Lymburner",
        "feedTitle": "High Scalability",
        "feed_id": "5b33c76cb2438d5708dc197e",
        "guid": "426227:4867632:36078004",
        "has_read": false,
        "link": "http://feedproxy.google.com/~r/HighScalability/~3/FFJQudxO2bU/open-source-database-ha-resources-from-severalnines.html",
        "publish_date": "2018-06-08T19:46:22.000Z",
        "timeAgo": "a month ago",
        "title": "Open Source Database HA Resources from Severalnines",
      },
    },
    "type": "FILTER_SET_DATA",
  },
]
`;
